src/http_rest_api_handler.cpp

```diff
@@ -670,12 +674,15 @@ Status HttpRestApiHandler::parseRequestComponents(HttpRequestComponents& request
                 return status;
             return StatusCode::OK;
         }
+        if (std::regex_match(request_path, sm, configReloadRegex)) {
```

Comment:
Why such change? Does order of ifs matter for performance?

Reply:
This is critical path. We shouldn't pay for additional regex_match while doing inference for KServe API.
It should be abysmal but not nonexistent.